src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -1646,6 +1646,27 @@ void M()
             }
         }
 
+        [Fact]
+        public void CS0221ERR_ConstOutOfRangeCheckedNullableInt()
+        {
+            // Confirm that we truncate the constant value before performing the range check
+            var template =
+@"public class C
+{
+    void M()
+    {
+        System.Console.WriteLine((int?)int.MinValue);
+        System.Console.WriteLine((int?)double.MinValue); //CS0221
+        System.Console.WriteLine((int?)int.MaxValue);
+        System.Console.WriteLine((int?)double.MaxValue); //CS0221
+    }
+}
+";
+
+            DiagnosticsUtils.VerifyErrorsAndGetCompilationWithMscorlib(template,
```

Comment:
We now recommend a better verification API: `CreateCompilation(template).VerifyDiagnostics(...)`. 
It produces an output that includes the diagnostic message and source at that location. That is friendlier for anyone reading the test.

It would look something like:
```
            var comp = CreateCompilation(text);
            comp.VerifyDiagnostics(
                // (22,15): error CS0217: In order to be applicable as a short circuit operator a user-defined logical operator ('MyClass.operator &(MyClass, MyClass)') must have the same return type and parameter types
                //       int i = f && f; // CS0217
                Diagnostic(ErrorCode.ERR_BadBoolOp, "f && f").WithArguments("MyClass.operator &(MyClass, MyClass)"));
```